opentelemetry-sdk/src/metrics/internal/aggregate.rs

```diff
@@ -94,11 +94,14 @@ impl<T: Number<T>> AggregateBuilder<T> {
     /// Wraps the passed in measure with an attribute filtering function.
     fn filter(&self, f: impl Measure<T>) -> impl Measure<T> {
         let filter = self.filter.clone();
-        move |n, mut attrs: AttributeSet| {
-            if let Some(filter) = &filter {
-                attrs.retain(filter.as_ref());
-            }
-            f.call(n, attrs)
+        move |n, attrs: &[KeyValue]| {
+            let filtered_attrs: Vec<KeyValue> = if let Some(filter) = &filter {
```

Comment:
Could we update it to this to avoid the allocation when there is no filter?

```rust
move |n, attrs: &[KeyValue]| {
    if let Some(filter) = &filter {
        let filtered_attrs: Vec<KeyValue> = attrs.iter().filter(|kv| filter(kv)).cloned().collect();
        f.call(n, &filtered_attrs);
    } else {
        f.call(n, attrs);
    };
}
```